std/json.d

```diff
@@ -355,40 +433,46 @@ struct JSONValue
             type_tag = JSON_TYPE.FLOAT;
             store.floating = arg;
         }
-        else static if (is(T : Value[Key], Key, Value))
+        else static if (!Policy.preserveMemberOrder && is(T : AAValue[Key], Key, AAValue))
         {
             static assert(is(Key : string), "AA key must be string");
             type_tag = JSON_TYPE.OBJECT;
-            static if (is(Value : JSONValue))
+            static if (is(AAValue : Value))
             {
-                JSONValue[string] t = arg;
+                JSONObject t = arg;
                 () @trusted { store.object = t; }();
             }
             else
             {
-                JSONValue[string] aa;
+                JSONObject aa;
                 foreach (key, value; arg)
-                    aa[key] = JSONValue(value);
+                    aa[key] = Value(value);
                 () @trusted { store.object = aa; }();
             }
         }
+        else static if (Policy.preserveMemberOrder && is(T == JSONObject))
```

Comment:
that seems restrictive compared to other policy which didn't only accept JSONObject but also things that converted to it via `Key : string` and `Value(value)`

Reply:
Maybe.  It seems a bit odd to take something that is unordered and then copy it to something that is ordered.  Not sure.

Reply:
An application could always do the conversion themselves to.  This is something that doesn't have to be implemented in the library (AFAICT).

Reply:
no, it could be AAordered!(Key, Value) with Key and Value convertible to destination types